File: backend/src/main/java/edu/hm/mobileappdev/repository/PlayerRepository.java
package edu.hm.mobileappdev.repository;

import edu.hm.mobileappdev.entity.Player;
import io.quarkus.hibernate.orm.panache.PanacheRepository;
import jakarta.enterprise.context.ApplicationScoped;
import java.util.Optional;
import java.util.UUID;

@ApplicationScoped
public class PlayerRepository implements PanacheRepository<Player> {
    public Player findById(UUID id) {
        return find("id", id).firstResult();
    }
    public Optional<Player> findByName(String playerName) {
        return find("name", playerName).firstResultOptional();
    }
}

File: lib/edu.hm.mobileappdev/repository/player_repository.dart
import 'package:mobileappdev/edu.hm.mobileappdev/repository/player.dart';

abstract class PlayerRepository {
  Future<void> addPlayer(String id);
  Future<Player> getPlayer(String id);
  Future<void> addPlayerWin(String id);
  Future<List<Player>> getPlayersList();
}